components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -356,6 +356,22 @@ export abstract class AbstractTypeORMWorkspaceDBImpl implements WorkspaceDB {
         return workspaceRepo.find({ ownerId: userId });
     }
 
+    public async getWorkspaceCountByCloneURL(
+        cloneURL: string,
+        sinceLastDays: number = 7,
+        type: string = "regular",
+    ): Promise<number> {
+        const workspaceRepo = await this.getWorkspaceRepo();
+        const since = new Date();
+        since.setDate(since.getDate() - sinceLastDays);
+        return workspaceRepo
+            .createQueryBuilder("ws")
+            .where('context->"$.repository.cloneUrl" = :cloneURL', { cloneURL })
+            .andWhere("creationTime > :since", { since: since.toISOString() })
+            .andWhere("type = :type", { type })
+            .getCount();
```

Comment:
Wow, many thanks for this clever fix! 👀

My only worry here is, how do we know that this query on all `d_b_workspace` entries won't be too slow/expensive? (Especially since we'll run it on every received webhook, which seems very frequent.)

Reply:
This is what EXPLAIN has to say:
<img width="1152" alt="Screenshot 2022-05-11 at 17 46 55" src="https://user-images.githubusercontent.com/372735/167892342-32de9071-c81a-41bb-a7f7-46573d493315.png">

Reply:
Aha, that totally addresses the performance concern then, right? (Or, do we need to somehow test/qualify this?)

Reply:
@svenefftinge, the number of rows sent is also a perf issue sometimes. 

If we're just checking for `count === 0` down below, we could start out with a `limit`ed version.

Reply:
I would expect typeorm to translate to count(*) when using `getCount`. Doesn't it do that?

Reply:
👍🏻 

https://github.com/typeorm/typeorm/blob/62518ae1226f22b2f230afa615532c92f1544f01/src/query-builder/SelectQueryBuilder.ts#L2849